src/connector/src/sink/doris_starrocks_connector.rs

```diff
@@ -337,3 +378,234 @@ impl InserterInner {
+        let mut request = Request::new(method, parsed_url);
+
+        if uri != self.url_load {
+            // Set timeout for non-load requests; load requests' timeout is controlled by `tokio::timeout`
+            *request.timeout_mut() = Some(self.stream_load_http_timeout);
+        }
+
+        let header = request.headers_mut();
+        for (k, v) in &self.header {
+            header.insert(
+                HeaderName::try_from(k)
+                    .map_err(|err| SinkError::DorisStarrocksConnect(err.into()))?,
+                HeaderValue::try_from(v)
+                    .map_err(|err| SinkError::DorisStarrocksConnect(err.into()))?,
+            );
+        }
+        header.insert(
+            "label",
+            HeaderValue::try_from(label)
+                .map_err(|err| SinkError::DorisStarrocksConnect(err.into()))?,
+        );
+
+        Ok(request)
+    }
+
+    pub fn build_begin_request_sender(&self, label: String) -> Result<MetaRequestSender> {
+        let request = self.build_request(self.url_begin.clone(), Method::POST, label)?;
+        Ok(MetaRequestSender::new(
+            self.client.clone(),
+            request,
+            self.fe_host.clone(),
+        ))
+    }
+
+    pub fn build_prepare_request_sender(&self, label: String) -> Result<MetaRequestSender> {
+        let request = self.build_request(self.url_prepare.clone(), Method::POST, label)?;
+        Ok(MetaRequestSender::new(
+            self.client.clone(),
+            request,
+            self.fe_host.clone(),
+        ))
+    }
+
+    pub fn build_commit_request_sender(&self, label: String) -> Result<MetaRequestSender> {
+        let request = self.build_request(self.url_commit.clone(), Method::POST, label)?;
+        Ok(MetaRequestSender::new(
+            self.client.clone(),
+            request,
+            self.fe_host.clone(),
+        ))
+    }
+
+    pub fn build_rollback_request_sender(&self, label: String) -> Result<MetaRequestSender> {
+        let request = self.build_request(self.url_rollback.clone(), Method::POST, label)?;
+        Ok(MetaRequestSender::new(
+            self.client.clone(),
+            request,
+            self.fe_host.clone(),
+        ))
+    }
+
+    pub async fn build_txn_inserter(&self, label: String) -> Result<InserterInner> {
+        let request = self.build_request(self.url_load.clone(), Method::PUT, label)?;
+        let mut request_for_redirection = request.try_clone().ok_or_else(|| {
+            SinkError::DorisStarrocksConnect(anyhow::anyhow!("Can't clone request"))
+        })?;
+
+        let resp = self
+            .client
+            .execute(request)
+            .await
+            .map_err(|err| SinkError::DorisStarrocksConnect(err.into()))?;
+
+        let be_url = try_get_be_url(&resp, self.fe_host.clone())?.ok_or_else(|| {
+            SinkError::DorisStarrocksConnect(anyhow::anyhow!("Can't get doris BE url",))
+        })?;
```

Comment:
Ditto.
```suggestion
        let be_url = try_get_be_url(&resp, self.fe_host.clone())?;
```

Reply:
By the way, why not do `try_get_be_url()` when the connector initialize and use the URL in later?

Reply:
This is because we send requests to FE, FE acts like a balancer and will redirect the request to a BE based on its internal balance algorithm, so we need to get the BE url for each request.